### PR TITLE
py-smart-open: Adding new version and dependencies

### PR DESCRIPTION
**Pull request by marcusboden**
My first a bit more complex dependency situation. Feel free to comment/corret.
The dependency handling changed quite a lot with newer versions of smart-open. I've decided to include all variants by default.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-smart-open/package.py [hunk 1]
```diff
@@ -13,14 +13,48 @@ class PySmartOpen(PythonPackage):
...
 
     depends_on('py-setuptools', type='build')
-    depends_on('py-requests', type=('build', 'run'))
-    depends_on('py-boto3', type=('build', 'run'))
-    depends_on('py-boto@2.3.2:', when='@1.8.4', type=('build', 'run'))
-    depends_on('py-google-cloud-storage', when='@1.10:', type=('build', 'run'))
-    depends_on('py-bz2file', when='^python@:2', type=('build', 'run'))
+
+    with when('@5:'):
+        depends_on('python@3.6:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-smart-open/package.py]:
```suggestion
        depends_on('python@3.6:3', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-smart-open/package.py [hunk 2]
```diff
@@ -13,14 +13,48 @@ class PySmartOpen(PythonPackage):
...
+
+        # google cloud support
+        variant(
+            'gcs', default=True, description='Builds a shared version of the library'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-smart-open/package.py]: Descriptions need to be updated

**Review by adamjstewart — CHANGES_REQUESTED** [review]
I would recommend making all variants False by default. This minimizes the number of dependencies that are built. Also, it's unlikely that someone will be using the same installation on both AWS and GCP.

**Comment by marcusboden** [comment]
I've updated the descriptions and changed the defaults for gcs, s3 and azure. But I kept the http as default, as I think that http is a pretty standard usecase and it only need py-requests.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**